internal/controller/datadogagent/component/agent/default.go

```diff
@@ -396,12 +401,17 @@ func agentOptimizedContainers(dda metav1.Object, requiredContainers []apicommon.
 }
 
 func coreAgentContainer(dda metav1.Object) corev1.Container {
+	autopilot := false
```

Comment:
these 4 lines repeat several times, have you considered extracting in a function?